syntax = "proto3";

message UserEvent {
    int32 user_id = 1;
    int32 item_id = 2;
    string timestamp = 3; // ISO-8601 string
    string action = 4;    // view / click / add_to_cart
}